File: app/src/main/java/de/xikolo/controllers/dialogs/ApiVersionDeprecatedDialog.java
package de.xikolo.controllers.dialogs;

import android.app.Dialog;
import android.content.DialogInterface;
import android.os.Bundle;
import android.support.annotation.NonNull;
import android.support.v7.app.AlertDialog;

import com.yatatsu.autobundle.AutoBundleField;

import java.text.DateFormat;
import java.util.Date;
import java.util.Locale;

import de.xikolo.R;
import de.xikolo.controllers.dialogs.base.BaseDialogFragment;

public class ApiVersionDeprecatedDialog extends BaseDialogFragment {

    public static final String TAG = ApiVersionDeprecatedDialog.class.getSimpleName();

    private ApiVersionDeprecatedDialogListener listener;

    @AutoBundleField Date deprecationDate;

    public void setDialogListener(ApiVersionDeprecatedDialogListener listener) {
        this.listener = listener;
    }

    @NonNull
    @Override
    public Dialog onCreateDialog(Bundle savedInstanceState) {
        DateFormat df = DateFormat.getDateInstance(DateFormat.LONG, Locale.getDefault());
        AlertDialog.Builder builder = new AlertDialog.Builder(getActivity(), R.style.AppTheme_Dialog)
                .setTitle(getString(R.string.dialog_api_deprecation_title, getString(R.string.app_name)))
                .setMessage(getString(R.string.dialog_api_deprecation_message, getString(R.string.app_name), df.format(deprecationDate)))
                .setPositiveButton(getString(R.string.dialog_api_deprecation_yes), new DialogInterface.OnClickListener() {
                    @Override
                    public void onClick(DialogInterface dialog, int which) {
                        if (listener != null) {
                            listener.onOpenPlayStoreClicked();
                        }
                    }
                })
                .setNegativeButton(getString(R.string.dialog_api_deprecation_no), new DialogInterface.OnClickListener() {
                    @Override
                    public void onClick(DialogInterface dialog, int which) {
                        if (listener != null) {
                            listener.onDismissed();
                        }
                    }
                })
                .setCancelable(true);

        AlertDialog dialog = builder.create();
        dialog.setCanceledOnTouchOutside(false);

        return dialog;
    }

    @Override
    public void onCancel(DialogInterface dialog) {
        if (listener != null) {
            listener.onDismissed();
        }
    }

    public interface ApiVersionDeprecatedDialogListener {
        void onOpenPlayStoreClicked();
        void onDismissed();
    }

}

File: app/src/main/java/de/xikolo/controllers/dialogs/ApiVersionDeprecatedDialog.kt
package de.xikolo.controllers.dialogs

import android.app.Dialog
import android.content.DialogInterface
import android.os.Bundle
import android.support.v7.app.AlertDialog
import com.yatatsu.autobundle.AutoBundleField
import de.xikolo.R
import de.xikolo.controllers.dialogs.base.BaseDialogFragment
import java.text.DateFormat
import java.util.*

class ApiVersionDeprecatedDialog : BaseDialogFragment() {

    companion object {
        @JvmField val TAG: String = ApiVersionDeprecatedDialog::class.java.simpleName
    }

    var listener: Listener? = null

    @AutoBundleField
    lateinit var deprecationDate: Date

    override fun onCreateDialog(savedInstanceState: Bundle?): Dialog {
        val df = DateFormat.getDateInstance(DateFormat.LONG, Locale.getDefault())
        val builder = AlertDialog.Builder(activity!!, R.style.AppTheme_Dialog)
            .setTitle(
                getString(
                    R.string.dialog_api_deprecation_title,
                    getString(R.string.app_name)
                )
            )
            .setMessage(
                getString(
                    R.string.dialog_api_deprecation_message,
                    getString(R.string.app_name),
                    df.format(deprecationDate)
                )
            )
            .setPositiveButton(getString(R.string.dialog_api_deprecation_yes)) { _, _ ->
                listener?.onOpenPlayStoreClicked()
            }
            .setNegativeButton(getString(R.string.dialog_api_deprecation_no)) { _, _ -> listener?.onDismissed() }
            .setCancelable(true)

        val dialog = builder.create()
        dialog.setCanceledOnTouchOutside(false)

        return dialog
    }

    override fun onCancel(dialog: DialogInterface?) {
        listener?.onDismissed()
    }

    interface Listener {
        fun onOpenPlayStoreClicked()
        fun onDismissed()
    }

}
